### PR TITLE
Added handling of account id, moved config to variables

### PR DESCRIPTION
**Pull request by peters-axon**

### CODE REVIEW

File: docusign-connector-product/README.md [hunk 1]
```diff
@@ -9,67 +16,102 @@ Axon Ivy’s [DocuSign](https://www.docusign.com/products/electronic-signature)
...
+    # If set, use a specific account id, otherwise use the default account of the user. (Probably only makes sense for JWT Token grant.)
+    accountId: ''
+
+    # Scope of grant.
+    scope: signature impersonation
+    
+    # Docusign base url for authentication.
+    baseUri: https://account-d.docusign.com/oauth
+
+    jwt:
```
Comment (ivy-rew) [inline docusign-connector-product/README.md:79]: love this change: much easier to digest the availables configs like this :+1:

File: docusign-connector-product/README.md [hunk 2]
```diff
@@ -9,67 +16,102 @@ Axon Ivy’s [DocuSign](https://www.docusign.com/products/electronic-signature)
...
+1. In the section `Authentication` click on `Generate RSA` in order to create a secure key-pair.  
  ![docusign-gen-rsa](images/authenticationGenerateRSA.png)
 
 1. Store the generated private key:
 	1. Copy the generated 'Private Key' to your clipboard
 	1. Save the changed application settings
 	1. Create a new empty text file called `docusign.pem` in your Designer 'configuration' directory
 	1. Paste the contents of your clipboard into the `docusign.pem` file
-	1. You can use another storage location for the pem file. Adjust the Rest Client Property `AUTH.systemKeyFile` to refer it. It should be a relative path to the 'configuration' directory or an absolute path on your system.
+	1. You can use another storage location for the pem file. Adjust the variable `jwt.keyFile` to refer it. It should be a relative path to the 'configuration' directory or an absolute path on your system.
```
Comment (ivy-rew) [inline docusign-connector-product/README.md]: please use the full qualified variable name:
`docusign-connector.jwt.keyFile`

... this is also how the variable is reachable in unit-tests and the engine-cockpit.

File: docusign-connector-product/README.md [hunk 3]
```diff
@@ -9,67 +16,102 @@ Axon Ivy’s [DocuSign](https://www.docusign.com/products/electronic-signature)
...
+	1. You can use another storage location for the pem file. Adjust the variable `jwt.keyFile` to refer it. It should be a relative path to the 'configuration' directory or an absolute path on your system.  
 ![docusign-pem](images/docuSignPem.png)
-![docusign-props](images/docuSignSystemProps.png)
 
 1. Define a user to act as service account:
 	1. Navigate to the `Users` overview and select your preferred service user
 	2. Copy the `API Username (id)` stated on the user detail page
-	3. Set it as value on the DocuSign Rest Clients property called `AUTH.systemUserId`
-![docusign-props](images/copyUserId.png)
+	3. Set it into the variable `jwt.userId`
```
Comment (ivy-rew) [inline docusign-connector-product/README.md]: qualify

File: docusign-connector-product/README.md [hunk 4]
```diff
@@ -9,67 +16,102 @@ Axon Ivy’s [DocuSign](https://www.docusign.com/products/electronic-signature)
...
+	
+1. JWT will be used automatically for processes run by the system user. If you want to use it
+   in general, set variable `jwt.use` to `true`
```
Comment (ivy-rew) [inline docusign-connector-product/README.md]: qualify

File: docusign-connector/config/rest-clients.yaml [hunk 5]
```diff
@@ -8,12 +8,14 @@ RestClients:
...
     - com.axonivy.connector.docusign.auth.OAuth2Feature
     Properties:
-      AUTH.integrationKey: your-app-key
-      AUTH.secretKey: ${decrypt:}
-      AUTH.systemUserId: ''
-      AUTH.systemKeyFile: docusign.pem
-      AUTH.scope: signature impersonation
-      AUTH.baseUri: https://account-d.docusign.com/oauth
+      AUTH.integrationKey: ${ivy.var.docusign-connector.integrationKey}
+      AUTH.secretKey: ${ivy.var.docusign-connector.secretKey}
```
Comment (ivy-rew) [inline docusign-connector/config/rest-clients.yaml]: this is a secret and should be decrypted: unfortunately there is not much tooling to do it but a workaround:
- add a password -property to the rest-client using the rest-client-editor
- fill the variable name ${ivy.var.xxx} as value

File: docusign-connector/src/com/axonivy/connector/docusign/event/EnvelopeCompleted.java [hunk 6]
```diff
@@ -22,73 +22,73 @@
 import ch.ivyteam.ivy.workflow.query.TaskQuery;
 
 public class EnvelopeCompleted extends AbstractProcessIntermediateEventBean {
-  public static UUID DOCU_SIGN_CLIENT_ID = UUID.fromString("3d3d7114-4397-48c9-8378-931fc35885c3");
+	public static UUID DOCU_SIGN_CLIENT_ID = UUID.fromString("3d3d7114-4397-48c9-8378-931fc35885c3");
```
Comment (ivy-rew) [inline docusign-connector/src/com/axonivy/connector/docusign/event/EnvelopeCompleted.java]: We should not change the formatting of the code for now reason. Makes it really hard to track changes if we added a bug or something in this change..

Please use our formatting prefs: and re-format the sources you changed
[eclipse-preferences.epf.zip](https://github.com/axonivy-market/docusign-connector/files/10740382/eclipse-preferences.epf.zip)
- Unzip this file
- In the Designer: File -> Import -> preferences > select the unpacked .epf
- Restart
- ... then you can apply auto-formatting

**Review by ivy-rew — APPROVED** [review]
Thanks for the re-furbishment; small but valuable changes :love_letter: 

Please address my remarks and we should be good to merge :+1:

File: docusign-connector-product/README.md (hunk 2 above)
Comment (peters-axon) [reply docusign-connector-product/README.md]: ok

File: docusign-connector-product/README.md (hunk 3 above)
Comment (peters-axon) [reply docusign-connector-product/README.md]: ok

File: docusign-connector/config/rest-clients.yaml (hunk 5 above)
Comment (peters-axon) [reply docusign-connector/config/rest-clients.yaml]: ok

**Review by ivy-rew — APPROVED** [review]
thanks a lot; feel free to merge :rocket:

**Pull request merged**